diff -ruN magplip.org/source/magplip_rev.h magplip/source/magplip_rev.h
--- magplip.org/source/magplip_rev.h	1998-04-01 16:55:36.000000000 +0200
+++ magplip/source/magplip_rev.h	2011-08-14 11:26:38.000000000 +0200
@@ -1,6 +1,6 @@
 #define	VERSION		38
 #define	REVISION	1
-#define	DATE	"1.4.98"
-#define	VERS	"magplip 38.1"
-#define	VSTRING	"magplip 38.1 (1.4.98)\r\n"
-#define	VERSTAG	"\0$VER: magplip 38.1 (1.4.98)"
+#define	DATE	"14.8.2011"
+#define	VERS	"magplip 38.1+"
+#define	VSTRING	"magplip 38.1+ (14.8.2011)\r\n"
+#define	VERSTAG	"\0$VER: magplip 38.1+ (14.8.2011)"
diff -ruN magplip.org/source/magplip_rev.i magplip/source/magplip_rev.i
--- magplip.org/source/magplip_rev.i	1998-04-01 16:55:36.000000000 +0200
+++ magplip/source/magplip_rev.i	2011-08-14 11:27:07.000000000 +0200
@@ -1,14 +1,14 @@
 VERSION		EQU	38
 REVISION	EQU	1
 DATE	MACRO
-		dc.b	'1.4.98'
+		dc.b	'3.6.2012'
 	ENDM
 VERS	MACRO
-		dc.b	'magplip 38.1'
+		dc.b	'magplip 38.1b'
 	ENDM
 VSTRING	MACRO
-		dc.b	'magplip 38.1 (1.4.98)',13,10,0
+		dc.b	'magplip 38.1b (3.6.2012)',13,10,0
 	ENDM
 VERSTAG	MACRO
-		dc.b	0,'$VER: magplip 38.1 (1.4.98)',0
+		dc.b	0,'$VER: magplip 38.1b (3.6.2012⁄)',0
 	ENDM
diff -ruN magplip.org/source/magport.asm magplip/source/magport.asm
--- magplip.org/source/magport.asm	1998-04-01 15:21:34.000000000 +0200
+++ magplip/source/magport.asm	2011-08-07 16:32:44.000000000 +0200
@@ -74,14 +74,19 @@
 ciab     equ     $bfd000
 BaseAX   equ     ciab+ciapra
 
+SETSELECT MACRO
+      bset     #CIAB_PRTRSEL,ciab+ciapra-BaseAX(a5)   ; raise PRTSEL line
+      ENDM
+
+CLRSELECT MACRO
+      bclr     #CIAB_PRTRSEL,ciab+ciapra-BaseAX(a5)   ; lower PRTSEL line
+      ENDM
 
 SETCIAOUTPUT MACRO
-      bset     #CIAB_PRTRSEL,ciab+ciapra-BaseAX(a5)   ; raise PRTSEL line
       st       ciaa+ciaddrb-BaseAX(\1)                ; data dir. => output
       ENDM
 
 SETCIAINPUT MACRO
-      bclr     #CIAB_PRTRSEL,ciab+ciapra-BaseAX(a5)   ; lower PRTSEL line
       sf       ciaa+ciaddrb-BaseAX(\1)                ; data dir. => input
       ENDM
 
@@ -164,6 +169,7 @@
 	 moveq    #CIAICRF_FLG,d0
 	 JSRLIB   AbleICR                             ; DISABLEINT
 	 lea      BaseAX,a5
+	 SETSELECT
 	 SETCIAOUTPUT a5
 	 move.b   (a5),d7                             ; SAMPLEINPUT, d7 = State
 	 move.l   a4,a3
@@ -203,6 +209,7 @@
 	 JSRLIB   SetICR                              ; CLEARINT
 	 move.w   #CIAICRF_FLG|CIAICRF_SETCLR,d0
 	 JSRLIB   AbleICR                             ; ENABLEINT
+	 CLRSELECT
 
 	 move.l   d2,d0                               ; return rc
 	 movem.l  (sp)+,d2-d7/a2-a6
@@ -238,6 +245,7 @@
 
 	 moveq    #CIAICRF_FLG,d0
 	 JSRLIB   AbleICR                             ; DISABLEINT
+	 SETSELECT
 
 	 move.b   (a5),d7                             ; SAMPLEINPUT
 	 cmp.b    #SYNCBYTE_HEAD,ciaa+ciaprb-BaseAX(a5) ; READCIABYTE
@@ -337,6 +345,7 @@
 	 JSRLIB   SetICR                              ; CLEARINT
 	 move.w   #CIAICRF_FLG|CIAICRF_SETCLR,d0
 	 JSRLIB   AbleICR                             ; ENABLEINT
+  CLRSELECT
 
 	 move.l   d5,d0                               ; return value
 	 movem.l  (sp)+,d2-d7/a2-a6
diff -ruN magplip.org/source/server.c magplip/source/server.c
--- magplip.org/source/server.c	1998-04-01 14:58:50.000000000 +0200
+++ magplip/source/server.c	2012-06-03 16:19:46.000000000 +0200
@@ -175,6 +175,7 @@
 #  define SETCIAOUTPUT
 #  define SETCIAINPUT
 #  define PARINIT(b)      ciaa.ciaddrb = 0xff; ciaa.ciaprb = 0x00
+#  define PAREXIT
 
 /* Good line : SELECT=POUT=PUSY=0 */
 #  define TESTLINE(b)     ((ciab.ciapra&(CIAF_PRTRSEL|CIAF_PRTRPOUT|CIAF_PRTRBUSY)) || pb->pb_Flags&PLIPF_RECEIVING)
@@ -187,8 +188,10 @@
 #  define SETCIAINPUT     ciab.ciapra &= ~CIAF_PRTRSEL; ciaa.ciaddrb = 0x00
 #  define PARINIT(b)      SETCIAINPUT;                                       \
 			ciab.ciaddra &= ~((b)->pb_HandshakeMask[HS_LINE]); \
-			ciab.ciaddra |= (b)->pb_HandshakeMask[HS_REQUEST]
-
+			ciab.ciaddra |= (b)->pb_HandshakeMask[HS_REQUEST] | CIAF_PRTRSEL
+#  define PAREXIT \
+			ciab.ciaddra &= ~(CIAF_PRTRSEL | CIAF_PRTRBUSY | CIAF_PRTRPOUT); \
+			ciab.ciapra  &= ~(CIAF_PRTRSEL | CIAF_PRTRBUSY | CIAF_PRTRPOUT)
 #  define TESTLINE(b)     (ciab.ciapra & (b)->pb_HandshakeMask[HS_LINE])
 #  define SETREQUEST(b)   ciab.ciapra |= (b)->pb_HandshakeMask[HS_REQUEST]
 #  define CLEARREQUEST(b) ciab.ciapra &= ~((b)->pb_HandshakeMask[HS_REQUEST])
@@ -274,6 +277,7 @@
 {
    if (pb->pb_AllocFlags & 4)
    {
+      PAREXIT;
       DISABLEINT;
       CLEARINT;
       RemICRVector(CIAABase, CIAICRB_FLG, &pb->pb_Interrupt);
diff -ruN magplip.org/source/smakefile magplip/source/smakefile
--- magplip.org/source/smakefile	1998-04-01 17:10:18.000000000 +0200
+++ magplip/source/smakefile	2011-08-05 20:51:19.000000000 +0200
@@ -17,7 +17,7 @@
 LIBS        = LIB LIB:amiga.lib LIB lib:sc.lib LIB lib:debug.lib
 CINCLUDES   = INCDIR="include:" INCDIR="netinclude:"
 ASMINCLUDES = INCDIR="INCLUDEASM:" $(CINCLUDES)
-CPUSUFFIX   = 020 # 000 010 020 030 040 060 ANY
+CPUSUFFIX   = 000 # 000 010 020 030 040 060 ANY
 CPUCCOPT    = CPU=68$(CPUSUFFIX)
 CP          = C:Copy
 MV          = C:Rename
@@ -31,8 +31,8 @@
 PROTECT     = protect
 BUILD_AMINET= /s/build-aminet
 ECHO        = Echo
-PROTOCOL    = LIN               # LIN or MAG
-PROTOCOL_LC = lin               # lin or mag (lowercase. stupid, I know... SZ)
+PROTOCOL    = MAG               # LIN or MAG
+PROTOCOL_LC = mag               # lin or mag (lowercase. stupid, I know... SZ)
 SUFFIX      = $(PRSUFFIX)$(CPUSUFFIX)
 #
 #####################################################################
@@ -160,14 +160,16 @@
    $(BUMPREV) $(VERSION) linplip
 
 clean_norm:
-   -@$(RM) *.o
-   -@$(RM) *.no
+   -@$(RM) "\#?.o"
+   -@$(RM) "\#?.no"
 
 clean_opt:
-   -@$(RM) *.o
-   -@$(RM) *.opt*
+   -@$(RM) "\#?.o"
+   -@$(RM) "\#?.opt\#?"
 
 clean: clean_norm clean_opt
+    -@$(RM) $(DEVICE_DIR)/$(NAME)
+    -@$(RM) $(DEVICE_DIR)/$(NAME).$(CPUSUFFIX)
 
 release:
    $(MAKE) CPUSUFFIX=000 PROTOCOL=MAG PROTOCOL_LC=mag all_opt
